core/src/main/scala/jms4s/config/config.scala

```diff
@@ -23,7 +23,21 @@ package jms4s.config
 
 import cats.Order
 
-sealed trait DestinationName        extends Product with Serializable
+import java.util.UUID
+
+sealed trait Destination
+
+sealed trait TemporaryDestination extends Destination
+
+case class TemporaryQueue(value: UUID = java.util.UUID.randomUUID()) extends TemporaryDestination {
```

Comment:
Creating an UUID is an impure and blocking operation, i will not make it a default value.
It should be executed inside a "blocking" block or, better, use https://typelevel.org/cats-effect/api/3.x/cats/effect/std/UUIDGen$.html.
This case-class should be a simple container of the name, the generation for the UUID should be moved elsewhere and executed in an appropriate context (an F[_] that has this capability).
If we want to maintain a default/utility method to create a `TemporaryQueue` with an UUID we can create a method(that works in `F[_]: UUIDGen` ) in the companion object .

Reply:
@faustin0 Good point! I have pushed a commit to try and improve this. Please let me know your thoughts...